src/probes.cpp

```diff
@@ -309,3 +357,94 @@ void context_entry(dyntracer_t* dyntracer, const RCNTXT* cptr) {
 
     state.exit_probe(Event::ContextEntry);
 }
+
+void environment_variable_define(dyntracer_t* dyntracer,
+                                 const SEXP symbol,
+                                 const SEXP value,
+                                 const SEXP rho) {
+  TracerState& state = tracer_state(dyntracer);
+  
+  state.enter_probe(Event::EnvironmentVariableDefine);
+  
+  if ((!state.assignment_stack_is_empty())
+  and (state.peek_assignment_stack().get_symbol() == symbol)
+  and (state.peek_assignment_stack().get_calling_environment() != rho)
+  and (type_of_sexp(value) == CLOSXP))    
+  {
+      sexptype_t type = state.peek_assignment_stack().get_type();
+
+      Call * assignment_call = state.get_parent_call(type, 1); 
+
+      assignment_call->set_dynamic_call();
+
+      if (state.peek_assignment_stack().get_calling_environment() != R_GlobalEnv) {
+        Function * parent_function = state.get_parent_call(type, 2)->get_function();
```

Comment:
The parent fucntion can have a different type. A Closure can call `->` for example.